include/opensm/osm_opensm.h

```diff
@@ -603,6 +641,39 @@ osm_opensm_wait_for_subnet_up(IN osm_opensm_t * p_osm, IN uint32_t wait_us,
 * SEE ALSO
 *********/
 
+/****f* OpenSM: OpenSM/osm_opensm_register_routing_engine
+ * NAME
+ *	osm_opensm_register_routing_engine
+ *
+ * DESCRIPTION
+ *	Register a new routing engine.
+ *
+ * SYNOPSIS
+ */
+cl_status_t osm_opensm_register_routing_engine(
```

Comment:
I added *osm_register_external_routing_engine* as a public function to osm_opensm.h

Reply:
Good suggestion, I changed the design accordingly.
Now, there is only **osm_register_external_routing_engine** as a public function.
external plugins can't specify their routing engine type. A new type will be generated by **osm_register_external_routing_engine** and it will be greater or equal than **OSM_ROUTING_ENGINE_TYPE_EXTERNAL**